node/src/components/consensus/highway_core/finality_detector/rewards.rs

```diff
@@ -154,29 +156,29 @@ mod tests {
         let mut state = State::new_test(&[Weight(5)], 0); // Alice is the only validator.
         let mut rng = TestRng::new();
 
-        // Round ID 0, length 16: Alice participates.
-        let p0 = add_vote!(state, rng, ALICE, 0, 4u8, 0x1; N)?; // Proposal
-        let w0 = add_vote!(state, rng, ALICE, 10, 4u8, None; p0)?; // Witness
+        // Round ID 0, length 32: Alice participates.
+        let p0 = add_vote!(state, rng, ALICE, 0, 5u8, 0x1; N)?; // Proposal
```

Comment:
Why did you have to increase the round exponent?

Reply:
Because `State::test_new` sets the minimum round exponent to 4, not 3, and this test included a decrease from 4 to 3. So the easiest fix — apart from also using `State::new` directly and adding a lot of boilerplate — was to just double all the numbers.

Reply:
So why did `State::test_new` sets the minimum to 4 and not 3?

Reply:
The test failed because I now `pre_validate_vote` in the tests, which was an omission in the first place.
`State::test_new` was already using 4, and if I change it to 3 now, other tests would fail.

I agree there's better ways to clean this up, but I wanted to get this out of the way and focus on HWY-156 first.